sql-expressions.md

```diff
@@ -0,0 +1,508 @@
+NOT` (not identical to) can be applied to any pair of values from a
+single data type, as well as some pairs of values from different data
+types.
+
+See also [this section over which data types are valid operands
+for each operator](functions-and-operators.html#operators).
+
+The following special rules apply:
+
+- `NULL` is always ordered smaller than every other value, even itself.
+- `NULL` is never equal to anything via `=`, even `NULL`. To check
+  whether a value is `NULL`, use the `IS` operator or the conditional
+  expression `IFNULL(..)`.
+
+### Set membership
+
+Syntax:
+
+	<expr> IN <expr>
+	<expr> IN ( ... subquery ... )
+
+	<expr> NOT IN <expr>
+	<expr> NOT IN ( ... subquery ... )
+
+Returns `TRUE` if and only if the value of the left operand is part of
+the result of evaluating the right operand.
+
+For example:
+
+	SELECT a IN (1, 2, 3) FROM sometable
+	SELECT a IN (SELECT * FROM allowedvalues) FROM sometable
+	SELECT ('x', 123) IN (SELECT * FROM rows)
+
+### String pattern matching
+
+Syntax:
+
+	<expr> LIKE <expr>
+	<expr> ILIKE <expr>
+	<expr> NOT LIKE <expr>
+	<expr> NOT ILIKE <expr>
+
+Evaluates both expressions as strings, then tests whether the string	 on the left
+matches the pattern given on the right. Returns `TRUE` if a match is found
+or `FALSE` otherwise, or the inverted value for the `NOT` variants.
+
+Patterns can contain `_` to match any single
+character, or `%` to match any sequence of zero or more characters.
+`ILIKE` causes the match to be tested case-insentively.
+
+For example:
+
+```sql
+SELECT 'monday' LIKE '%day' AS a, 'tuesday' LIKE 'tue_day' AS b, 'wednesday' ILIKE 'W%' AS c
++------+------+------+
+|  a   |  b   |  c   |
++------+------+------+
+| true | true | true |
++------+------+------+
+```
+
+### String Matching using POSIX regular expressions
+
+Syntax:
+
+	<expr> ~ <expr>
+	<expr> ~* <expr>
+	<expr> !~ <expr>
+	<expr> !~* <expr>
+
+Evaluates both expressions as strings, then tests whether the string	 on the left
+matches the pattern given on the right. Returns `TRUE` if a match is found
+or `FALSE` otherwise, or the inverted value for the  `!` variants.
+
+The pattern is expressed using
+[POSIX regular expression syntax](https://en.wikipedia.org/wiki/Regular_expression). Unlike
+`LIKE` patterns, a regular expression is allowed to match anywhere
+inside a string, not only at the beginning.
+
+### String Matching using SQL regular expressions
+
+Syntax:
+
+	<expr> SIMILAR TO <expr>
+	<expr> NOT SIMILAR TO <expr>
+
+Evaluates both expressions as strings, then tests whether the string	 on the left
+matches the pattern given on the right. Returns `TRUE` if a match is found
+or `FALSE` otherwise, or the inverted value for the `NOT`  variant.
+
+The pattern is expressed using the SQL standard's definition of a regular expression.
+This is a mix of SQL `LIKE` patterns and POSIX regular expressions:
+
+- `_` and `%` denote any character or any string, respectively;
```

Comment:
The whole makes a single paragraph. I do not understand why the need to split it; the semicolons are breaking the reading already as appropriate.